sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -113,7 +113,17 @@ object functions {
    * @group normal_funcs
    * @since 1.3.0
    */
-  def lit(literal: Any): Column = typedLit(literal)
+  def lit(literal: Any): Column = literal match {
+    case c: Column => c
```

Comment:
Can you pass a column as a literal? if so, does it work this way currently?

Reply:
Yep. I just inlined `typedlit(literal)` and changed `Column(Literal.create(literal))` to `Column(Literal(literal))`. I don't expect there is any behavior difference.

Reply:
Oh OK. That just seems odd that it's allowed? a literal that isn't a literal

Reply:
Agree it's odd. I don't know the context either 😄